# Remove the obsoletes line in kernel-headers
# Add provides for kernel-devel so there's no conflict
# and fix akmod-nvidia

diff --git a/scripts/package/mkspec b/scripts/package/mkspec
index 7c477ca7d..1158f5559 100755
--- a/scripts/package/mkspec
+++ b/scripts/package/mkspec
@@ -25,0 +26 @@ fi
+PROVIDES_DRM=""
@@ -27 +28 @@ if grep -q CONFIG_DRM=y .config; then
-	PROVIDES=kernel-drm
+	PROVIDES_DRM="Provides: kernel-drm = %{version}"
@@ -30 +30,0 @@ fi
-PROVIDES="$PROVIDES kernel-$KERNELRELEASE"
@@ -51 +51,3 @@ $S	Source: kernel-$__KERNELRELEASE.tar.gz
-	Provides: $PROVIDES
+	$PROVIDES_DRM
+	Provides: kernel = %{version}
+	Provides: kernel-uname-r = %{version}
+	Provides: installonlypkg(kernel) = %{version}
@@ -61 +63 @@ $S	Source: kernel-$__KERNELRELEASE.tar.gz
-	Obsoletes: kernel-headers
+	Provides: installonlypkg(kernel) = %{version}
@@ -72,0 +75,3 @@ $S$M	Group: System Environment/Kernel
+$S$M	Provides: kernel-devel = %{version}
+$S$M	Provides: kernel-devel-uname-r = %{version}
+$S$M	Provides: installonlypkg(kernel) = %{version}
